tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2397,9 +2407,47 @@ Status ConvertExpandDims(OpConverterParams* params) {
   // ExpandDim's ability to add an axis at end of the shape.
   int trt_axis;
   TF_RETURN_IF_ERROR(ConvertAxis(axis[0], dims.nbDims + 1, node_def.name(),
-                                 /*use_implicit_batch=*/true, &trt_axis));
+                                 params->use_implicit_batch, &trt_axis));
   if (params->validation_only) return Status::OK();
 
+#if IS_TRT_VERSION_GE(6, 0, 0, 0)
```

Comment:
It would be nice if we can refactor this to share code with Converter::SqueezeTensor:
(1) a common routine to decide whether input_dims has a dynamic value
(2) a routine for ConcatenateAndReshape(absl::span<nvinfer1::ITensor const*> values, nvinfer1::ITensor* new_shape)

Reply:
Yes, it is a good idea. There are other ops which might need similar functionalities: many ops that change shape need special methods in dynamic shape mode. I will review a few other cases, and I will improve this.

Reply:
(1) Using common HasStaticShape routine to decide on whether the tensor has static or dynamic shapes.
(2) Added SliceShape routine to take slices of a tensor's shape and combine it into a new shape tensor.
(3) Added separate DynamicExpandDims routine which will also be used by the Pack converter.